crates/rspack_plugin_footer/src/lib.rs

```diff
@@ -0,0 +1,46 @@
+use async_trait::async_trait;
+use rspack_core::{
+  rspack_sources::{BoxSource, ConcatSource, RawSource, SourceExt},
+  Plugin,
+};
+
+#[derive(Debug)]
+pub struct FooterPlugin {
+  pub options: FooterPluginOptions,
+}
+#[derive(Debug)]
+pub struct FooterPluginOptions {
+  pub footer: String,
+}
+impl FooterPlugin {
+  pub fn new(options: FooterPluginOptions) -> Self {
+    Self { options }
+  }
+}
+#[async_trait]
+impl Plugin for FooterPlugin {
+  fn name(&self) -> &'static str {
+    "rspack.FooterPlugin"
+  }
+  async fn process_assets_stage_additions(
```

Comment:
Can you add information or examples on using hooks as well?

Reply:
you mean js side or rust side?

Reply:
rust side